x-pack/plugins/fleet/server/routes/output/handler.ts

```diff
@@ -142,3 +144,23 @@ export const deleteOutputHandler: RequestHandler<
     return defaultIngestErrorHandler({ error, response });
   }
 };
+
+export const postLogstashApiKeyHandler: RequestHandler = async (context, request, response) => {
+  const esClient = context.core.elasticsearch.client.asCurrentUser;
+  try {
+    if (!(await canCreateLogstashApiKey(esClient))) {
```

Comment:
(nit) Chaining a negation directly onto an `await` call like this adds some syntax noise w/ the parentheses, etc. This might be cleaner to pull out into a variable, e.g. 

```ts
const hasCreatePrivileges = await canCreateLogstashApiKey(esClient)

If (!hasCreatePrivileges) { 
  throw new FleetUnauthorizedError('Missing permissions to create logstash API key');
}
```